cleanrl/ppo_dna_atari_envpool.py

```diff
@@ -0,0 +1,426 @@
+            advantages, _ = compute_advantages(
+                rewards, dones, values, next_done, next_value, args.gamma, args.policy_gae_lambda
+            )
+            _, returns = compute_advantages(rewards, dones, values, next_done, next_value, args.gamma, args.value_gae_lambda)
+
+        # flatten the batch
+        b_obs = obs.reshape((-1,) + envs.single_observation_space.shape)
+        b_logprobs = logprobs.reshape(-1)
+        b_actions = actions.reshape((-1,) + envs.single_action_space.shape)
+        b_advantages = advantages.reshape(-1)
+        b_returns = returns.reshape(-1)
+        b_values = values.reshape(-1)
+
+        # Policy network optimization
+        agent_policy.train()
+        b_inds = np.arange(args.batch_size)
+        clipfracs = []
+        for epoch in range(args.policy_update_epochs):
+            np.random.shuffle(b_inds)
+            for start in range(0, args.batch_size, args.policy_batch_size):
+                end = start + args.policy_batch_size
+                mb_inds = b_inds[start:end]
+
+                _, newlogprob, entropy, _, _ = agent_policy.get_action_and_value(b_obs[mb_inds], b_actions.long()[mb_inds])
+                logratio = newlogprob - b_logprobs[mb_inds]
+                ratio = logratio.exp()
+
+                with torch.no_grad():
+                    # calculate approx_kl http://joschu.net/blog/kl-approx.html
+                    old_approx_kl = (-logratio).mean()
+                    approx_kl = ((ratio - 1) - logratio).mean()
+                    clipfracs += [((ratio - 1.0).abs() > args.clip_coef).float().mean().item()]
+
+                mb_advantages = b_advantages[mb_inds]
+                if args.norm_adv:
+                    mb_advantages = (mb_advantages - mb_advantages.mean()) / (mb_advantages.std() + 1e-8)
+
+                # Policy loss
+                pg_loss1 = -mb_advantages * ratio
+                pg_loss2 = -mb_advantages * torch.clamp(ratio, 1 - args.clip_coef, 1 + args.clip_coef)
+                pg_loss = torch.max(pg_loss1, pg_loss2).mean()
+
+                entropy_loss = entropy.mean()
+                policy_loss = pg_loss - args.ent_coef * entropy_loss
+
+                policy_optimizer.zero_grad()
+                policy_loss.backward()
+                nn.utils.clip_grad_norm_(agent_policy.parameters(), args.max_grad_norm)
+                policy_optimizer.step()
+
+            if args.target_kl is not None:
+                if approx_kl > args.target_kl:
+                    break
+
+        # Value network optimization
+        agent_value.train()
+        for epoch in range(args.value_update_epochs):
+            np.random.shuffle(b_inds)
+            for start in range(0, args.batch_size, args.value_batch_size):
+                end = start + args.value_batch_size
+                mb_inds = b_inds[start:end]
+
+                newvalue = agent_value.get_value(b_obs[mb_inds])
+
```

Comment:
I hit a major regression with the latest script. It appears it's because `newvalue` here is not flattened

<img width="1362" alt="image" src="https://user-images.githubusercontent.com/5555347/203342912-5d770322-b656-4792-bc09-0fd6986e97f9.png">